### PR TITLE
WASM: Threads part 1

### PR DESCRIPTION
**Pull request by Cellule**
Implements part of #3477
- Support for Shared WebAssembly.Memory
- Support for atomic load/store
- Add new ListOpnd to allow Instr to have multiple dst and/or srcs to track liveness. Currently supports only RegOpnd

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/3919)

### CODE REVIEW

File: lib/Backend/AsmJsJITInfo.cpp [hunk 1]
```diff
@@ -78,6 +78,12 @@ AsmJsJITInfo::GetWasmSignatureAddr(uint index) const
     Assert(index < m_data.wasmSignatureCount);
     return m_data.wasmSignaturesBaseAddr + index * sizeof(Wasm::WasmSignature);
 }
+
+bool AsmJsJITInfo::IsSharedMemory() const
+{
+    return !!m_data.wasmIsSharedMemory;
```
Comment (MikeHolman) [inline lib/Backend/AsmJsJITInfo.cpp]: nit: for consistency would prefer `!= FALSE`

**Comment by Cellule** [comment]
@agarwal-sandeep @sigatrev You guys might be interested to look at ListOpnd which should be usefull for ARM64

**Comment by Cellule** [comment]
@leirocks @akroshg Could one of you guys review the changes for the Atomics implementation?

**Comment by Cellule** [comment]
I am going to split this into separate PR as much as I can to simplify code review

**Pull request closed without merging**